bob-backend/src/pearl/holder.rs

```diff
@@ -430,6 +427,50 @@ impl Holder {
     }
 }
 
+#[async_trait::async_trait]
+impl BloomProvider<Key> for Holder {
+    type Filter = <Storage<Key> as BloomProvider<Key>>::Filter;
+    async fn check_filter(&self, item: &Key) -> FilterResult {
+        let storage = self.storage().read().await;
+        if let Some(storage) = &storage.storage {
+            return BloomProvider::check_filter(storage, item).await;
+        }
+        FilterResult::NeedAdditionalCheck
+    }
+
+    fn check_filter_fast(&self, _item: &Key) -> FilterResult {
+        FilterResult::NeedAdditionalCheck
+    }
+
+    async fn offload_buffer(&mut self, needed_memory: usize, level: usize) -> usize {
+        let mut storage = self.storage().write().await;
+        match &mut storage.storage {
```

Comment:
I suggest to replace with `if let Some(st) = storage.storage.as_ref_mut() {...} else { 0 }